home/web/client/providers/rpc.js

```diff
@@ -0,0 +1,118 @@
+import { isArray, isObject, isEqual, compact } from 'lodash';
+import { isBigNumber } from 'web3/lib/utils/utils';
+import { toPromise } from 'dapps-react-components/src/util/util';
+import logger from 'dapps-react-components/src/util/logger';
+import * as actions from '../actions/rpc';
+
+export default class RpcProvider {
+
+  state = {};
+  delay = 500;
+
+  constructor (store, web3) {
+    this.store = store;
+    this.web3 = web3;
+    this.tickArr = this.getTickArr();
+  }
+
+  init () {
+    this.refreshTick();
```

Comment:
It might be a good idea to do first `eth_accounts` call synchronously (but it's just a suggestion for future PR)